test/LinalgExt/tiling-to-tile-op-e2e.mlir

```diff
@@ -0,0 +1,46 @@
+// RUN: mlir-proto-opt %s -linalg-ext-tiling-to-tile-op="tile-size=10" | FileCheck %s
+
+// CHECK-LABEL: reverse_1d_tensor(
+//  CHECK-SAME:   %[[T:[0-9a-z]+]]: tensor<?x?xf32>
+func @reverse_1d_tensor(%arg0 : tensor<?x?xf32>) -> tensor<?x?xf32> {
+  %c0 = arith.constant 0 : index
+  %c1 = arith.constant 1 : index
+  %d0 = tensor.dim %arg0, %c0 : tensor<?x?xf32>
+  %d1 = tensor.dim %arg0, %c1 : tensor<?x?xf32>
+  %init = linalg.init_tensor [%d0, %d1] : tensor<?x?xf32>
+
+  //      CHECK: %[[C10:.*]] = arith.constant 10 : index
+  //      CHECK:   %[[I:.*]] = linalg.init_tensor {{.*}} : tensor<?x?xf32>
+  //      CHECK: linalg_ext.tile %[[C10]] outs(%[[I]]: tensor<?x?xf32>) -> (tensor<?x?xf32>) {
+  //      CHECK: ^bb0(%[[OFF:.*]]: index, %[[SZ:.*]]: index, %[[I_ITER:.*]]: tensor<?x?xf32>):
+  //      CHECK:   %[[in:.*]] = tensor.extract_slice %[[T]]{{.*}} : tensor<?x?xf32> to tensor<?x?xf32>
+  //      CHECK:   %[[out:.*]] = tensor.extract_slice %[[I_ITER]]{{.*}} : tensor<?x?xf32> to tensor<?x?xf32>
+  //      CHECK:   %[[RES:.*]] = linalg_ext.reverse dimensions(dense<0> : tensor<1xi64>) 
+  // CHECK-SAME:     ins(%[[in]] : tensor<?x?xf32>) outs(%[[out]] : tensor<?x?xf32>) : tensor<?x?xf32>
+  //      CHECK:   linalg_ext.tile_yield %[[RES]] : tensor<?x?xf32>
```

Comment:
Huh, I would expect the reverse op to _not_ be tileable in our understanding of it. It seems like you want the reverse of the last tile to update the place occupied by the first tile?

Reply:
if you are not doing in-place updates, the reverse op is tilable.

Reply:
I would say it is tilable but not "inplace".
Bufferization could not happen inplace here, this is similar to what a transposition would be: not inplace (see https://github.com/google/iree-llvm-sandbox/tree/main/python/examples/transpose).